lib/mongo/auth/aws/credentials_retriever.rb

```diff
@@ -83,14 +83,20 @@ def credentials
             return credentials
           end
 
+          credentials = ecs_metadata_credentials
```

Comment:
It looks like `ecs_metadata_credentials` returns nil whenever the http request fails and in that case the driver will fallback to `ec2_metadata_credentials`. But from my reading of the ["Obtaining Credentials" section of the spec](https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#obtaining-credentials), the driver should only ever attempt one of these. In other words, I think the logic should be:
```
if ENV['AWS_CONTAINER_CREDENTIALS_RELATIVE_URI']
    credentials = ecs_metadata_credentials
else
    credentials = ec2_metadata_credentials
fi
```

Reply:
Looking at the auth spec again it seems to mandate both behaviors. I filed https://jira.mongodb.org/browse/DRIVERS-995 to clarify the situation.

The behavior implemented in this PR supports all possible deployment scenarios, including ECS tasks authenticating with instance role credentials.

Reply:
I believe all other drivers have implemented the order I described above so Ruby should probably follow suit for consistency. 

> including ECS tasks authenticating with instance role credentials.

I'm not familiar with this case. Can you explain what the credential lookup flow would be and why my above suggest would not support it?

Reply:
On second thought, I don't want to hold up your work. We can continue this discussion in Jira.